File: backend/app/schemas/balance_audit.py
from datetime import date, datetime
from decimal import Decimal
from typing import Any, Dict

from pydantic import BaseModel, ConfigDict


class BalanceAuditBase(BaseModel):
    date: date
    balances: Dict[str, float | None]  # Allow None for robustness
    debts: Decimal
    owed: Decimal
    net_position: Decimal


class BalanceAuditCreate(BaseModel):
    date: date
    # Fields are optional to support server-side calculation
    balances: Dict[str, float | None] | None = None
    debts: Decimal | None = None
    owed: Decimal | None = None
    net_position: Decimal | None = None


class BalanceAuditResponse(BalanceAuditBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: SlaveOfCapitalism/SlaveOfCapitalism/Models/BalanceAudit.swift
import Foundation

struct BalanceAuditResponse: Codable, Identifiable, Sendable {
    let id: Int
    let date: String
    let balances: [String: Double?]
    let debts: Decimal
    let owed: Decimal
    let netPosition: Decimal
    let createdAt: String
    let updatedAt: String
}

struct BalanceAuditCreate: Codable, Sendable {
    let date: String
    var balances: [String: Double?]?
    var debts: Decimal?
    var owed: Decimal?
    var netPosition: Decimal?
}
